src/layers/CanvasLayer.ts

```diff
@@ -0,0 +1,36 @@
+import { select } from 'd3-selection';
+import Layer from './Layer';
+import {
+  OnMountEvent,
+  OnUpdateEvent,
+} from '../interfaces';
+
+abstract class CanvasLayer extends Layer {
+  ctx: CanvasRenderingContext2D;
+  elm: HTMLElement;
+
+  onMount(event: OnMountEvent) {
+    super.onMount(event);
+    this.elm = event.elm;
+
+    const canvas = select(event.elm).append('canvas')
```

Comment:
Do we want d3 select in other that svg layer? Should we have some simpler alternative or just use plain js?

Reply:
used plain js